File: PaymentServer/WebbappResponse.cs
using System;
using System.IO;
using System.Net;
using System.Collections.Generic;
using System.Text;

namespace PaymentServer
{
	class WebbappResponse
	{
		HttpListenerResponse http_response;

		public WebbappResponse(HttpListenerResponse response)
		{
			http_response = response;
		}

		public void send(string msg)
		{
			http_response.AppendHeader("Access-Control-Allow-Origin", "*");
			Stream output = http_response.OutputStream;
			output.Write(Encoding.UTF8.GetBytes(msg));
			output.Dispose();

			//send
			http_response.Close();
		}
	}
}

File: PaymentServerJava/src/paymentserver/WebbappResponse.java
package paymentserver;

import com.sun.net.httpserver.Headers;
import com.sun.net.httpserver.HttpExchange;

import java.io.*;

public class WebbappResponse
{
	private HttpExchange http_exchange;

	public WebbappResponse(HttpExchange http_exchange)
	{
		this.http_exchange = http_exchange;
	}

	public final void send(String msg)
	{
	    try
		{
			OutputStreamWriter output = new OutputStreamWriter(http_exchange.getResponseBody());
			output.write(msg);


			Headers h = http_exchange.getResponseHeaders();
			h.clear();
			h.add("Access-Control-Allow-Origin", "*");
			h.add("Access-Control-Allow-Methods", "PUT,POST,GET,OPTIONS,DELETE");

			http_exchange.sendResponseHeaders(200, msg.length());

			output.flush();
			output.close();
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}